Userland/Libraries/LibWeb/HTML/NavigationType.h

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (c) 2023, Andrew Kaster <akaster@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/String.h>
+
+// FIXME: Generate this from the IDL file that just has an enum in it
+namespace Web::Bindings {
+enum class NavigationType {
+    Push,
+    Replace,
+    Reload,
+    Traverse,
+};
+
+inline String idl_enum_to_string(NavigationType value)
+{
+    switch (value) {
+    case NavigationType::Push:
+        return MUST(String::from_utf8("Push"sv));
```

Comment:
Oops, my comment didn't drop, I meant to say here - maybe this can just return `"Push"_string`

Reply:
Ah, good catch! I'll fix that one up in the morning